statement/src/_literal/_language_tag.rs

```diff
@@ -76,13 +76,17 @@ impl std::cmp::PartialEq<LangTag<'_>> for &str {
     }
 }
 
+impl std::cmp::Ord for LangTag<'_> {
+    fn cmp(&self, other: &Self) -> std::cmp::Ordering {
+        self.0
+            .to_ascii_lowercase()
```

Comment:
Maybe we should normalize language tag casing, having the equality allocate is quite a footgun

Even better, there is `eq_ignore_ascii_case`:


```suggestion
    self.0.eq_ignore_ascii_case(&other.0)
```

Reply:
I'm reluctant to internally normalize language tags. While the case is not significant to the data model, users tend to care about it.

That being said, +1 to avoid allocations and use `eq_ignore_ascii_case` instead!

Reply:
In fact, the PR already uses `eq_ignore_ascii_case` for equality, but comparison is more tricky. We could avoid the allocation by implementing a corresponding `cmp_ignore_ascii_case`, there is an example [here](https://stackoverflow.com/a/63871901/1235487).

Reply:
done in b7813e0